pandas/io/clipboard.py

```diff
@@ -75,21 +74,22 @@ def to_clipboard(obj, excel=None, sep=None, **kwargs):  # pragma: no cover
       - OS X:
     """
     from pandas.util.clipboard import clipboard_set
-    if excel is None:
-        excel = True
 
-    if excel:
-        try:
-            if sep is None:
-                sep = '\t'
-            buf = StringIO()
-            obj.to_csv(buf, sep=sep, **kwargs)
-            clipboard_set(buf.getvalue())
-            return
-        except:
-            pass
-
-    if isinstance(obj, DataFrame):
+    # COMPAT: allow None for `excel` and `sep`
+    if excel in (True, None) and isinstance(obj, (DataFrame, Series)):
+        if 'sep' not in kwargs or not kwargs['sep']:
+            kwargs['sep'] = '\t'
+        if 'line_terminator' not in kwargs:
+            import os
+            kwargs['line_terminator'] = os.linesep
+        buf = StringIO()
+        obj.to_csv(buf, **kwargs)
+        objstr = buf.getvalue()
+    elif isinstance(obj, DataFrame):
+        if 'sep' in kwargs:
+            if kwargs['sep']:
+                raise ValueError("sep kw is unsupported with excel=False")
```

Comment:
maybe i should remove `ValueError` here? just delete `sep` kw (if it's `None`) for backwards compatibility. And let `to_string` to raise error otherwise.